core/src/main/scala/chisel3/Data.scala

```diff
@@ -372,6 +374,11 @@ abstract class Data extends HasId with NamedComponent with SourceInfoDoc {
     }
   }
 
+  // Denotes if this is a probe type
+  private var _probeInfo:         ProbeInfo = ProbeInfo(false, false)
```

Comment:
This may be better modeled as an `private var _isProbe: Option[ProbeInfo]` where `ProbeInfo` is changed to:

```scala
case class ProbeInfo(writable: Boolean)
```

This is more of a nit because the `ProbeInfo(false, false)` and `ProbeInfo(false, true)` cases are both the same right now.

Reply:
i like that, thanks for the suggestion!

Reply:
While it's less important for Probes since they're less common than binding, I'd rather use `private var _probeInfoVar: ProbeInfo = null` with option boxing
```
private[chisel3] def probeInfo: Option[ProbeInfo] = Option(_probeInfoVar)
```
This will save 12-bytes per Probe with the same type safety (as long as no one calls `_probeInfoVar)`

Same thing as done elsewhere: https://github.com/chipsalliance/chisel/blob/0ce115e9be85f0556d453857a439186f6e1d9d74/core/src/main/scala/chisel3/Data.scala#L399